shared/configHelper.js

```diff
@@ -33,141 +33,114 @@ var path = require('path'),
 
     var processorList = null;
 
-    switch (command || '') {
-    case 'version':
-        logInfo(forcecli.name + ' version ' + SDK.version);
+    switch (commandName || '') {
+    case SDK.commands.version.name:
+        printVersion(cli);
         process.exit(0);
         break;
-    case 'create': 
-        processorList = createArgsProcessorList(forcecli, false); 
-        break;
-    case 'createWithTemplate': 
-        processorList = createArgsProcessorList(forcecli, true); 
+    case SDK.commands.create.name: 
+    case SDK.commands.createWithTemplate.name: 
+        processorList = createArgsProcessorList(cli, commandName);
         break;
     default:
-        usage(forcecli);
+        usage(cli);
         process.exit(1);
     };
 
     commandLineUtils.processArgsInteractive(commandLineArgs, processorList, handler);
 }
 
-function usage(forcecli) {
-    var forcecliName = forcecli.name;
-    var forcecliVersion = SDK.version;
-    var appTypes = forcecli.appTypes;
-    var platforms = forcecli.platforms;
+function printVersion(cli) {
+    logInfo(cli.name + ' version ' + SDK.version);
+}
+
+function printArgs(cli, commandName) {
+    getArgsExpanded(cli, commandName)
+        .filter(arg => !arg.hidden)
+        .forEach(arg => logInfo('    ' + (!arg.required  ? '[' : '') + '--' + arg.name + '=' + arg.description + (!arg.required ? ']' : ''), COLOR.magenta));
+}    
+
+function usage(cli) {
+    var cliName = cli.name;
+    var cliVersion = SDK.version;
+    var appTypes = cli.appTypes;
+    var platforms = cli.platforms;
     
-    logInfo('\n' + forcecliName + ': ' + forcecli.description, COLOR.cyan);
+    logInfo('\n' + cliName + ': ' + cli.description, COLOR.cyan);
     logInfo('\nUsage:\n', COLOR.cyan);
-    logInfo(forcecliName + ' create', COLOR.magenta);
-    if (appTypes.length > 1) {
-        logInfo('    --apptype=<Application Type> (' + appTypes.join(', ') + ')', COLOR.magenta);
-    }
-    if (platforms.length > 1) {
-        logInfo('    --platform=<Comma separated plaforms> (' + platforms.join(', ') + ')', COLOR.magenta);
-    }
-    logInfo('    --appname=<Application Name>', COLOR.magenta);
-    logInfo('    --packagename=<App Package Identifier> (e.g. com.mycompany.myapp)', COLOR.magenta);
-    logInfo('    --organization=<Organization Name> (Your company\'s/organization\'s name)', COLOR.magenta);
-    if (appTypes.indexOf('hybrid_remote') >= 0) {
-        logInfo('    --startpage=<App Start Page> (The start page of your remote app. Only required for hybrid_remote)', COLOR.magenta);
-    }
-    logInfo('    [--outputdir=<Output directory> (Leave empty for current directory)]', COLOR.magenta);
-    logInfo('\n OR \n', COLOR.cyan);
-    logInfo(forcecliName + ' createWithTemplate', COLOR.magenta);
-    if (platforms.length > 1) {
-        logInfo('    --platform=<Comma separated plaforms> (' + platforms.join(', ') + ')', COLOR.magenta);
+    for (var i=0; i<cli.commands.length; i++) {
```

Comment:
All the possible params etc are in constants.js. So the code left here is simply looping over them.